src/core/network/client.ts

```diff
@@ -82,6 +82,23 @@ function includesMedia(payload: Record<string, unknown>) {
   })
 }
 
+// Types are pretty wrong if applied to a Partial but
+// is passable if applied to an already defined object
+// E.g. compactOptions({ a: number, b?: number })
+// produces Type { a: number }
+type CompactKeys<T> = Exclude<
+  { [K in keyof T]: T[K] extends {} ? K : never }[keyof T],
+  undefined
+>
+type Compact<T> = Pick<T, CompactKeys<T>>
+function compactOptions<T>(options: T): Compact<T> {
+  const compactKeys = Object.keys(options).filter(
+    (key) => key !== undefined
```

Comment:
You're filtering for undefined keys, did you mean to filter by `key => options[key] !== undefined`? Note that `key` is unfortunately `string` so type casting will become necessary. The function does not work yet.

Reply:
Oof. Good catch!